distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Upload_Table.enso

```diff
@@ -24,44 +27,82 @@ from project.Internal.Result_Set import result_set_to_table
 ## PRIVATE
    Creates a new database table with the provided structure and returns the name
    of the created table.
-create_table_structure connection table_name structure primary_key temporary allow_existing on_problems =
-    case table_name.is_nothing.not && all_known_table_names connection . contains table_name of
+
+   The user-facing function that handles the dry-run logic.
+create_table_implementation connection table_name structure primary_key temporary allow_existing on_problems = Panic.recover SQL_Error <|
+    case all_known_table_names connection . contains table_name of
         True ->
-            if allow_existing then table_name else Error.throw (Table_Already_Exists.Error table_name)
+            if allow_existing then connection.query (SQL_Query.Table_Name table_name) else Error.throw (Table_Already_Exists.Error table_name)
         False ->
-            effective_table_name = resolve_effective_table_name table_name temporary
-            aligned_structure = align_structure structure
-            if aligned_structure.is_empty then Error.throw (Illegal_Argument.Error "An empty table cannot be created: the `structure` must consist of at list one column description.") else
-                resolved_primary_key = resolve_primary_key aligned_structure primary_key
-                create_table_statement = prepare_create_table_statement connection effective_table_name aligned_structure resolved_primary_key temporary on_problems
-                update_result = create_table_statement.if_not_error <|
-                    connection.execute_update create_table_statement
-                update_result.if_not_error <|
-                    effective_table_name
+            dry_run = Context.Output.is_enabled.not
+            effective_table_name = case dry_run of
+                False -> table_name
+                True -> dry_run_table_name table_name
+            effective_temporary = case dry_run of
+                False -> temporary
+                True -> True
+            created_table_name = Context.Output.with_enabled <|
+                connection.jdbc_connection.run_within_transaction <|
+                    if dry_run then
+                        # If the temporary table already exists, we drop it and will overwrite it.
+                        connection.drop_table table_name if_exists=True
+                    internal_create_table_structure connection effective_table_name structure primary_key effective_temporary on_problems
+            case dry_run of
+                False -> connection.query (SQL_Query.Table_Name created_table_name)
+                True ->
+                    created_table = connection.internal_allocate_hidden_table created_table_name
+                    warning = Dry_Run_Operation.Warning "The Output context is disabled, so only a dry-run of `create_table` was performed. Instead of the actual table, a temporary table called "+created_table_name.pretty+" has been created."
```

Comment:
```suggestion
                    warning = Dry_Run_Operation.Warning "Only a dry run of `create_table` has occurred, creating a temporary table ("+created_table_name.pretty+") instead of the actual."
```
More aligned with the other warnings.